addons/web/static/tests/core/condition_tree_tests.js

```diff
@@ -243,41 +243,121 @@ QUnit.test("domainFromExpression", function (assert) {
 
 QUnit.test("expressionFromTree", function (assert) {
     const options = {
-        getFieldDef: (name) => (name === "x" ? {} /** any field */ : null),
+        getFieldDef: (name) => {
+            if (["foo", "bar"].includes(name)) {
+                return {}; // any field
+            }
+            if (["foo_ids", "bar_ids"].includes(name)) {
+                return { type: "many2many" };
+            }
+            return null;
+        },
     };
     const toTest = [
         {
-            expressionTree: condition("x", "=", false),
-            result: `not x`,
+            expressionTree: condition("foo", "=", false),
+            result: `not foo`,
         },
         {
-            expressionTree: condition("x", "=", false, true),
-            result: `x`,
+            expressionTree: condition("foo", "=", false, true),
+            result: `foo`,
         },
         {
-            expressionTree: condition("x", "!=", false),
-            result: `x`,
+            expressionTree: condition("foo", "!=", false),
+            result: `foo`,
         },
         {
-            expressionTree: condition("x", "!=", false, true),
-            result: `not x`,
+            expressionTree: condition("foo", "!=", false, true),
+            result: `not foo`,
         },
         {
             expressionTree: condition("y", "=", false),
             result: `not "y"`,
         },
         {
-            expressionTree: condition("x", "between", [1, 3]),
-            result: `x >= 1 and x <= 3`,
+            expressionTree: condition("foo", "between", [1, 3]),
+            result: `foo >= 1 and foo <= 3`,
         },
         {
-            expressionTree: condition("x", "between", [1, expression("uid")], true),
-            result: `not ( x >= 1 and x <= uid )`,
+            expressionTree: condition("foo", "between", [1, expression("uid")], true),
+            result: `not ( foo >= 1 and foo <= uid )`,
         },
         {
             expressionTree: complexCondition("uid"),
             result: `uid`,
         },
+        {
+            expressionTree: condition("foo_ids", "in", []),
+            result: `set(foo_ids).intersection()`,
+        },
+        {
+            expressionTree: condition("foo_ids", "in", [1]),
+            result: `set(foo_ids).intersection([1])`,
```

Comment:
Can you add `set` in `IGNORED_IN_EXPRESSION` and a small test in `TestValidationTools`